lib/drivers/medtronic600/medtronic600Simulator.js

```diff
@@ -128,6 +128,39 @@ class Medtronic600Simulator {
     }
 
     if (this.currBasal != null) {
+      // Back-fill Auto-Basal gaps
+      if (this.currBasal.scheduleName === 'Auto-Basal') {
+        const autoBasalDifference = Date.parse(event.time).valueOf() -
+          Date.parse(this.currBasal.time).valueOf();
+        // Auto-basal microboluses occur roughly every 5 minutes, but no more than 6 minutes apart.
+        // The pump doesn't tell us if the algorithm chose *not* to micro-bolus.
+        // If a 6 minute gap appears between micro-boluses, there must have been a gap.
+        if (autoBasalDifference > 360000) {
+          this.currBasal
+            .set('duration', 300000)
+            .set('basalEndTime', new Date(Date.parse(this.currBasal.time).valueOf() + 300000).toISOString());
+          const currBasalIndex = this.currBasal.index;
+          this.events.push(this.currBasal.done());
+
+          const insertedBasalDate = new Date(Date.parse(this.currBasal.time).valueOf() + 300000);
+          const insertedBasal = this.config.builder.makeScheduledBasal()
+            .with_time(insertedBasalDate.toISOString())
+            .with_deviceTime(sundial.formatDeviceTime(
+              sundial.parseFromFormat(this.currBasal.deviceTime).valueOf() + 300000))
+            .with_rate(0)
+            .with_scheduleName('Auto-Basal')
+            .with_duration(Date.parse(event.time) - Date.parse(this.currBasal.time) - 300000)
+            .with_conversionOffset(this.currBasal.conversionOffset)
+            .with_clockDriftOffset(this.currBasal.clockDriftOffset)
+            .with_timezoneOffset(this.currBasal.timezoneOffset)
+            .set('index', currBasalIndex + 300000)
```

Comment:
Why can't we use TZOUtil to set the offsets here? And if we're not using TZOUtil, why are we setting `index` and `jsDate`?